src/ai/defaultai.cc

```diff
@@ -1931,14 +1930,14 @@ void DefaultAI::update_buildable_field(BuildableField& field) {
 		}
 		assert(field_owner == pn);  // It is us
 
-		consider_own_msites(location, field, any_connected_imm, any_unconnected_imm);
+		consider_own_msites(location, field, any_imm_connected_to_wh, any_imm_not_connected_to_wh);
 
 	} while (second_area.advance(map));
 
 	assert(field.military_loneliness <= 1000);
 
-	if (any_unconnected_imm && any_connected_imm && field.military_in_constr_nearby == 0) {
-		field.unconnected_nearby = true;
+	if (any_imm_not_connected_to_wh && any_imm_connected_to_wh && field.military_in_constr_nearby == 0) {
```

Comment:
if we ever use this we might need to reconsider the condition here. but for now it is ok.

Reply:
yes, I will review all this later.